packages/labs/motion/src/flip.ts

```diff
@@ -0,0 +1,584 @@
+    let dScaleX = 1;
+    let dScaleY = 1;
+    if (ancestorProps !== undefined) {
+      // gather scaling data for ancestors
+      ancestorProps.forEach((a) => {
+        if (a.width) {
+          dScaleX = dScaleX / (a.width as number);
+        }
+        if (a.height) {
+          dScaleY = dScaleY / (a.height as number);
+        }
+      });
+      // Move position by ancestor scaling amount.
+      if (from.left !== undefined && to.left !== undefined) {
+        from.left = dScaleX * (from.left as number);
+        to.left = dScaleX * (to.left as number);
+      }
+      if (from.top !== undefined && to.top !== undefined) {
+        from.top = dScaleY * (from.top as number);
+        to.top = dScaleY * (to.top as number);
+      }
+    }
+    return {from, to};
+  }
+
+  protected calculateFrames(from: CSSValues, to: CSSValues, center = false) {
+    const fromFrame: Keyframe = {};
+    const toFrame: Keyframe = {};
+    let hasFrames = false;
+    const props: CSSValues = {};
+    for (const p in to) {
+      const f = from[p],
+        t = to[p];
+      if (p in transformProps) {
+        const tp = transformProps[p as keyof typeof transformProps];
+        if (f === undefined || t === undefined) {
+          continue;
+        }
+        const op = tp(f as number, t as number);
+        if (op.transform !== undefined) {
+          props[p] = op.value!;
+          hasFrames = true;
+          fromFrame.transform = `${fromFrame.transform ?? ''} ${op.transform}`;
+        }
+      } else if (f !== t && f !== undefined && t !== undefined) {
+        hasFrames = true;
+        fromFrame[p] = f;
+        toFrame[p] = t;
+      }
+    }
+    fromFrame.transformOrigin = toFrame.transformOrigin = center
+      ? 'center center'
+      : 'top left';
+    this.flipProps = props;
+    return hasFrames ? [fromFrame, toFrame] : undefined;
+  }
+
+  protected async animate(
+    frames: Keyframe[] | undefined,
+    options = this.options.animationOptions
+  ) {
+    this.frames = frames;
+    if (this.isAnimating() || this.isDisabled()) {
+      return false;
+    }
+    if (this.options.onFrames) {
+      this.frames = frames = this.options.onFrames(this);
+      this.log('modified frames', frames);
+    }
+    if (frames === undefined) {
+      return false;
+    }
+    this.log('animate', [frames, options]);
+    this.animation = this.element.animate(frames, options);
+    const controller = this.getController();
+    controller?.add(this);
+    try {
+      await this.animation.finished;
+    } catch (e) {
+      // cancelled.
+    }
+    controller?.remove(this);
+    return true;
+  }
+
+  protected isAnimating() {
+    return this.animation?.playState === 'running' || this.animation?.pending;
+  }
+
+  log(message: string, data?: unknown) {
+    if (this.shouldLog && !this.isDisabled()) {
+      console.log(message, this.options.id, data);
+    }
+  }
+}
+
+export const flip = directive(Flip);
```

Comment:
This jsdocs should go here, since this is the symbol exported.

Reply:
Added